### ISSUE TITLE
[Event Request] record 9001 "User Group Member" - OnBeforeModifyUserGroupMembership

### ISSUE THREAD

**Issue by nybragaramos** [microsoft/ALAppExtensions#13762]
```
[IntegrationEvent(false, false)]
local procedure OnBeforeModifyUserGroupMembership(var UserGroupMember: Record "User Group Member"; xUserGroupMember: Record "User Group Member"; var IsHandled: Boolean)
begin
end;

local procedure ModifyUserGroupMembership()
var
    //------> NEW
    IsHandled: Boolean;
begin
    IsHandled := false;
     OnBeforeModifyUserGroupMembership(Rec, xRec, IsHandled)
    if IsHandled then
        exit;
    //<------ NEW
end;
```

**Comment by AlexanderYakunin** [comment]
Please explain why do you need this event with possibility to override normal user membership modification?

**Comment by nybragaramos** [comment]
I'm building a workflow where permission is needed to change a user group membership. Without this event I have no way of avoiding the user group change process.

**Comment by AlexanderYakunin** [comment]
Sorry, due to security reasons we have no plans to provide events in user management objects. If you have full scenario, we can discuss how it can be implemented.

**Comment by nybragaramos** [comment]
I am producing a regulatory application. Since this application will be used within the pharmaceutical and food industries, where user data and its respective permission are especially sensitive, changing a user from one group to another must only occur after an authorization process through a workflow.  At the moment there is no way to directly prevent the change, with this event I can leave the change on hold to be accepted only after workflow approval.

**Comment by db-bc** [comment]
Are the page(s) where you can change group extensible or subscribable? If so, you might be able subscribe to the attempted change of group (validation/modify/whatever), write whatever 'pending' records you need for the workflow, then commit and error to cancel the 'real' change. Then continue with your workflow and convert the 'pending' change to a 'real' one later.

This of course wouldn't cover modifications via code, PowerShell, etc. - but it might be possible to handle all user-visible ways to change group membership. I say "might" because I haven't tried it personally.

**Comment by RomanZanizdra** [comment]
Hi @nybragaramos . Did the previous suggestion help you? Can we close this issue?

**Comment by darjoo** [comment]
I'll be closing this as no response and seems to be an old issue.
@nybragaramos If it hasn't been resolved, please do reopen and let us know.
